test/e2e/apimachinery/webhook.go

```diff
@@ -1819,12 +1821,17 @@ func testCRDDenyWebhook(f *framework.Framework) {
 	ginkgo.By("Creating a custom resource definition that should be denied by the webhook")
 	name := fmt.Sprintf("e2e-test-%s-%s-crd", f.BaseName, "deny")
 	kind := fmt.Sprintf("E2e-test-%s-%s-crd", f.BaseName, "deny")
-	group := fmt.Sprintf("%s-crd-test.k8s.io", f.BaseName)
-	apiVersions := []apiextensionsv1beta1.CustomResourceDefinitionVersion{
+	group := fmt.Sprintf("%s-crd-test.example.com", f.BaseName)
+	apiVersions := []apiextensionsv1.CustomResourceDefinitionVersion{
 		{
 			Name:    "v1",
 			Served:  true,
 			Storage: true,
+			Schema: &apiextensionsv1.CustomResourceValidation{
+				OpenAPIV3Schema: &apiextensionsv1.JSONSchemaProps{
+					Type: "object",
```

Comment:
not sure if it matters for this test, but if you don't want to prune all fields, `XPreserveUnknownFields: true`

Reply:
Hit this in the tests too. I've added `XPreserveUnknownFields: true`.